src/js/components/FileInput/FileInput.js

```diff
@@ -85,6 +87,46 @@ const FileInput = forwardRef(
     const removeRef = useRef();
     const RemoveIcon = theme.fileInput.icons.remove;
 
+    formContext.useFormField(
+      maxSize
+        ? {
+            name,
+            validate: (fileList) => {
+              let numOfFiles = 0;
+              let message;
+              for (let i = 0; i < fileList.length; i += 1) {
+                const file = fileList[i];
+                if (file.size > maxSize) {
```

Comment:
maxSize id documented as a number or object which specifies a number and a string for units. This seems to be expecting a number of bytes. How does the object get converted to bytes? How will it appear in validation messages then?

Reply:
Down the line and in response to the `PropTypes` comment as well, currently this will only accept maxSize in bytes but down the line I will be looking into adding a more verbose way to specify maxSize to look something like:

```js
const maxSize = {{
  unit: 'mb',
  size: '5'
}}
```

But if in our first pass we won't be looking to go into it I would be removing the human-readable format for byte parsing before anything else gets published later.